fri/src/two_adic_pcs.rs

```diff
@@ -24,167 +24,135 @@ use tracing::{info_span, instrument};
 use crate::verifier::{self, FriError};
 use crate::{prover, FriConfig, FriProof};
 
-/// We group all of our type bounds into this trait to reduce duplication across signatures.
-pub trait TwoAdicFriPcsGenericConfig: Default {
-    type Val: TwoAdicField;
-    type Challenge: TwoAdicField + ExtensionField<Self::Val>;
-    type Challenger: FieldChallenger<Self::Val>
-        + GrindingChallenger<Witness = Self::Val>
-        + CanObserve<<Self::FriMmcs as Mmcs<Self::Challenge>>::Commitment>
-        + CanSample<Self::Challenge>;
-    type Dft: TwoAdicSubgroupDft<Self::Val>;
-    type InputMmcs: 'static
-        + for<'a> DirectMmcs<Self::Val, Mat<'a> = RowMajorMatrixView<'a, Self::Val>>;
-    type FriMmcs: DirectMmcs<Self::Challenge>;
+pub struct TwoAdicFriPcs<Val, Dft, InputMmcs, FriMmcs> {
+    // degree bound
```

Comment:
Actually I had a degree bound because I was planning to do decomposition automatically, but ended up changing back to manual decomposition of polys that are too large (decomposition is through the `Domain`).

so maybe I can get rid of it